src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -422,20 +427,19 @@ void IExtensionConfigProvider.Initialize(ExtensionConfigContext context)
         private void InitializeLinuxLogging()
         {
             // Determine host platform
-            bool inLinuxDedicated = this.platformInformationService.InLinuxAppService();
-            bool inLinuxConsumption = this.platformInformationService.InLinuxConsumption();
+            bool inConsumption = this.PlatformInformationService.IsInConsumptionPlan();
 
-            string tenant = this.platformInformationService.GetLinuxTenant();
-            string stampName = this.platformInformationService.GetLinuxStampName();
-            string containerName = this.platformInformationService.GetContainerName();
+            string tenant = this.PlatformInformationService.GetLinuxTenant();
+            string stampName = this.PlatformInformationService.GetLinuxStampName();
+            string containerName = this.PlatformInformationService.GetContainerName();
 
             // If running in linux, initialize the EventSource listener with the appropiate logger.
             LinuxAppServiceLogger linuxLogger = null;
-            if (inLinuxDedicated)
+            if (!inConsumption)
             {
                 linuxLogger = new LinuxAppServiceLogger(writeToConsole: false, containerName, tenant, stampName);
             }
-            else if (inLinuxConsumption)
+            else if (inConsumption)

```

Comment:
nit: the if clause in this else-if is not necessary. 

In addition, below this, we have a guarantee that linuxLogger is initialized so we don't need the null check.